compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -417,37 +418,37 @@ private void defineConstructs(BLangPackage pkgNode, SymbolEnv pkgEnv) {
         // Define type definitions.
         this.typePrecedence = 0;
 
-        // Treat constants and type definitions in the same manner, since constants can be used as
-        // types. Also, there can be references between constant and type definitions in both ways.
-        // Thus visit them according to the precedence.
-        List<BLangNode> typeAndClassDefs = new ArrayList<>();
-        pkgNode.constants.forEach(constant -> typeAndClassDefs.add(constant));
-        pkgNode.typeDefinitions.forEach(typDef -> typeAndClassDefs.add(typDef));
-        List<BLangClassDefinition> classDefinitions = getClassDefinitions(pkgNode.topLevelNodes);
-        classDefinitions.forEach(classDefn -> typeAndClassDefs.add(classDefn));
+        // Treat constants, type definitions and xmlns declarations in the same manner, since constants can be
+        // used as types. Also, there can be references between constant, type definitions and xmlns declarations in
+        // both ways. Thus visit them according to the precedence.
```

Comment:
I would say visit them together since constants are also types and can be referred to in XMLNS declarations. Additionally, types and XMLNS declarations can also be used in constants.

Reply:
`Additionally, types and XMLNS declarations can also be used in constants.` isn't this part implied by this `Also, there can be references between constant, type definitions and xmlns declarations in
        // both ways`

Reply:
Yeah, either way is OK. The first sentence needs to change though, since only the XMLNS bit is changed and not the rest of the sentence. Thus, the suggestion to change the second one also.